The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10994701571/job/30523855463) [(plain)](https://github.com/rust-lang/rust/commit/1478a183159970ed334b07a087522e39c6424163/checks/30523855463/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.677 Building wheels for collected packages: reuse
#13 2.678   Building wheel for reuse (pyproject.toml): started
#13 2.920   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.921   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 2.922   Stored in directory: /tmp/pip-ephem-wheel-cache-ye6z12z3/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.924 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.322 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.322 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.848 Collecting virtualenv
#13 3.848 Collecting virtualenv
#13 3.885   Downloading virtualenv-20.26.5-py3-none-any.whl (6.0 MB)
#13 3.975      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.0/6.0 MB 68.5 MB/s eta 0:00:00
#13 4.032 Collecting platformdirs<5,>=3.9.1
#13 4.036   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.055 Collecting distlib<1,>=0.3.7
#13 4.066      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 93.3 MB/s eta 0:00:00
#13 4.100 Collecting filelock<4,>=3.12.2
#13 4.109   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.109   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.192 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.381 Successfully installed distlib-0.3.8 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.26.5
#13 DONE 4.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      194496 kB
DirectMap2M:     9242624 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/a772336fb3fbd1fe4493077fcfe04e0221296a99/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-a772336fb3fbd1fe4493077fcfe04e0221296a99-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
+use crate::spec::{Target, TargetOptions, base};
 
 pub(crate) fn target() -> Target {
-
     Target {
         // LLVM 15 doesn't support OpenHarmony yet, use a linux target instead.
         llvm_target: "loongarch64-unknown-linux-musl".into(),
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/run-make/extern-fn-struct-passing-abi/rmake.rs" "/checkout/compiler/rustc_target/src/spec/targets/hexagon_unknown_linux_musl.rs" "/checkout/tests/run-make/extern-fn-struct-passing-abi/test.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm_unknown_linux_musleabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv8m_main_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv6m_nuttx_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_ohos.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_win7_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64gc_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_haiku.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv8m_main_none_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64imac_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64gc_unknown_hermit.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_wrs_vxworks.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7a_kmc_solid_asp3_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7r_none_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/avr_unknown_gnu_atmega328.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_fuchsia.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32im_risc0_zkvm_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/hexagon_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_none_softfloat.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv8m_main_nuttx_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7em_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv8m_base_nuttx_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/mips_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/xtensa_esp32_espidf.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_linux_muslspe.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32gc_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_unknown_none.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/armebv7r_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/msp430_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv6_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_linux_gnux32.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv5te_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7_unknown_linux_uclibceabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32ima_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/bpfel_unknown_none.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_sony_psx.rs" "/checkout/compiler/rustc_target/src/spec/targets/sparc64_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_uwp_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unikraft_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_linux_gnuspe.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_wrs_vxworks.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32imac_esp_espidf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_apple_tvos_sim.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_apple_darwin.rs" "/checkout/compiler/rustc_target/src/spec/targets/mips_unknown_linux_uclibc.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32imac_unknown_nuttx_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_sony_psp.rs" "/checkout/compiler/rustc_target/src/spec/targets/sparc_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/armebv7r_none_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/bpfeb_unknown_none.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Mon Sep 23 13:00:39 UTC 2024
  network time: Mon, 23 Sep 2024 13:00:39 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_target/src/spec/mod.rs
+++ b/compiler/rustc_target/src/spec/mod.rs
@@ -1886,6 +1886,7 @@ supported_targets! {
 
     ("aarch64-unknown-linux-ohos", aarch64_unknown_linux_ohos),
     ("armv7-unknown-linux-ohos", armv7_unknown_linux_ohos),
+    ("loongarch64-unknown-linux-ohos", loongarch64_unknown_linux_ohos),
     ("x86_64-unknown-linux-ohos", x86_64_unknown_linux_ohos),
 
     ("x86_64-unknown-linux-none", x86_64_unknown_linux_none),

--- a/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_ohos.rs
+++ b/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_ohos.rs
@@ -1,0 +1,24 @@
+use crate::spec::{Target, TargetOptions, base};
+
+pub(crate) fn target() -> Target {
+    Target {
+        // LLVM 15 doesn't support OpenHarmony yet, use a linux target instead.
+        llvm_target: "loongarch64-unknown-linux-musl".into(),
+        metadata: crate::spec::TargetMetadata {
+            description: Some("LoongArch64 OpenHarmony".into()),
+            tier: Some(3),
+            host_tools: Some(false),
+            std: Some(true),
+        },
+        pointer_width: 64,
+        data_layout: "e-m:e-p:64:64-i64:64-i128:128-n32:64-S128".into(),
+        arch: "loongarch64".into(),
+        options: TargetOptions {
+            cpu: "generic".into(),
+            features: "+f,+d".into(),
+            llvm_abiname: "lp64d".into(),
+            max_atomic_width: Some(64),
+            ..base::linux_ohos::opts()
+        },
+    }
+}

--- a/src/doc/rustc/src/platform-support.md
+++ b/src/doc/rustc/src/platform-support.md
@@ -315,6 +315,7 @@ target | std | host | notes
 `i686-uwp-windows-msvc` | ✓ |  | [^x86_32-floats-return-ABI]
 [`i686-win7-windows-msvc`](platform-support/win7-windows-msvc.md) | ✓ |   | 32-bit Windows 7 support [^x86_32-floats-return-ABI]
 [`i686-wrs-vxworks`](platform-support/vxworks.md) | ✓ |  | [^x86_32-floats-return-ABI]
+[`loongarch64-unknown-linux-ohos`](platform-support/openharmony.md) | ✓ |   | LoongArch64 OpenHarmony
 [`m68k-unknown-linux-gnu`](platform-support/m68k-unknown-linux-gnu.md) | ? |  | Motorola 680x0 Linux
 `mips-unknown-linux-gnu` | ✓ | ✓ | MIPS Linux (kernel 4.4, glibc 2.23)
 `mips-unknown-linux-musl` | ✓ |  | MIPS Linux with musl 1.2.3

--- a/src/doc/rustc/src/platform-support/openharmony.md
+++ b/src/doc/rustc/src/platform-support/openharmony.md
@@ -2,6 +2,14 @@
 
 **Tier: 2**
 
+* aarch64-unknown-linux-ohos
+* armv7-unknown-linux-ohos
+* x86_64-unknown-linux-ohos
+
+**Tier: 3**
+
+* loongarch64-unknown-linux-ohos
+
 Targets for the [OpenHarmony](https://gitee.com/openharmony/docs/) operating
 system.
 

--- a/tests/assembly/targets/targets-elf.rs
+++ b/tests/assembly/targets/targets-elf.rs
@@ -252,6 +252,9 @@
 //@ revisions: loongarch64_unknown_linux_musl
 //@ [loongarch64_unknown_linux_musl] compile-flags: --target loongarch64-unknown-linux-musl
 //@ [loongarch64_unknown_linux_musl] needs-llvm-components: loongarch
+//@ revisions: loongarch64_unknown_linux_ohos
+//@ [loongarch64_unknown_linux_ohos] compile-flags: --target loongarch64-unknown-linux-ohos
+//@ [loongarch64_unknown_linux_ohos] needs-llvm-components: loongarch
 //@ revisions: loongarch64_unknown_none
 //@ [loongarch64_unknown_none] compile-flags: --target loongarch64-unknown-none
 //@ [loongarch64_unknown_none] needs-llvm-components: loongarch

